db/models.js

```diff
@@ -6,6 +6,27 @@ exports.fetchTopics = () => {
   });
 };
 
+exports.fetchArticles = (sort_by) => {
+  let queryStr =
+    "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, (SELECT COUNT (*) FROM comments WHERE comments.article_id = articles.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id";
+  const sortableColumns = ["created_at"];
+
+  if (sort_by) {
+    if (sortableColumns.includes(sort_by)) {
+      queryStr += ` ORDER BY ${sort_by} DESC`;
+    } else {
+      return Promise.reject({ status: 400, message: "Invalid sort_by." });
+    }
+  }
+
+  return db.query(queryStr).then((result) => {
+    result.rows.forEach((article) => {
+      article.comment_count = Number(article.comment_count);
+    });
```

Comment:
This can be done by adjusting your SQL statement to type the response as an integer.